vendor/github.com/yannh/statuspage-go-sdk/component.go

```diff
@@ -5,7 +5,7 @@ type Component struct {
 	Description        *string `json:"description,omitempty"`
 	GroupID            *string `json:"group_id,omitempty"`
 	Showcase           *bool   `json:"showcase,omitempty"`
-	Status             *string `json:"status,omitempty"`
+	Status             string  `json:"status,omitempty"`
```

Comment:
I think this is not how you intended, `omitempty` only works in this case if the pointer is `nil`, but not if the string is empty.
In previous case if you set `status = ""`, it will still submit an empty request to Statuspage API, which will always set the status to `operational`